src/org/ses/android/soap/utils/PreferencesManager.java

```diff
@@ -69,5 +77,58 @@ public static void removeFingerprint(Context context) {
 
     }
 
+    // Take an arraylist of Cacheable things and save to preferences.
+    public static void saveCacheableList(Context context, String saveName,
+                                         ArrayList<Cacheable> objectList) {
+        SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        JSONArray arrayString = new JSONArray();
+        for (Cacheable object : objectList) {
+            JSONObject objectString = object.toJSON();
+            arrayString.put(objectString);
+        }
+        SharedPreferences.Editor editor = mPreferences.edit();
+        editor.putString(saveName, arrayString.toString());
+        editor.commit();
+    }
+
+    public static void clearCacheable(Context context, String saveName) {
+        SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = mPreferences.edit();
+        if (mPreferences.getString(saveName, "").length() > 0)
+        {
+            editor.remove(saveName);
+            editor.commit();
+        }
+    }
+
+    // Try to receive Cacheable list. Need to upcast your Cacheable using constructor after retrieving.
+    public static ArrayList<Cacheable> getCacheableList(Context context, String saveName) {
+        SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        String objectString = mPreferences.getString(saveName, "");
+        ArrayList<Cacheable> objectList = new ArrayList<Cacheable>();
+
+        if (objectString.equals("")) {
+            return null;
+        }
+
+        JSONArray objectJSONArray;
+        try {
+            objectJSONArray = new JSONArray(objectString);
+        } catch (JSONException e) {
+            return null;
+        }
+
+        for (int i=0; i < objectJSONArray.length(); i++) {
+            try {
+                Cacheable object = new Cacheable(objectJSONArray.getJSONObject(i));
+                objectList.add(object);
+            } catch (JSONException e) {
+                // Move on to the next one
```

Comment:
Can you do insert a `Log.w` so we know that there's an error? Might help uncover subtle bugs.